kroxylicious-kms-provider-aws-kms/src/test/java/io/kroxylicious/kms/provider/aws/kms/AwsKmsTest.java

```diff
@@ -0,0 +1,187 @@
+                    .withCauseInstanceOf(UnknownAliasException.class)
+                    .withMessageContaining("key 'alias' is not found");
+        });
+    }
+
+    @Test
+    void resolveAliasInternalServerError() {
+        withMockAwsWithSingleResponse(null, 500, kms -> {
+            var aliasStage = kms.resolveAlias("alias");
+            assertThat(aliasStage)
+                    .failsWithin(Duration.ofSeconds(5))
+                    .withThrowableThat()
+                    .withCauseInstanceOf(KmsException.class)
+                    .withMessageContaining("Operation failed");
+        });
+    }
+
+    @Test
+    void generateDekPair() {
+        // response from https://docs.aws.amazon.com/kms/latest/APIReference/API_GenerateDataKey.html
+        var response = """
+                {
+                  "CiphertextBlob": "AQEDAHjRYf5WytIc0C857tFSnBaPn2F8DgfmThbJlGfR8P3WlwAAAH4wfAYJKoZIhvcNAQcGoG8wbQIBADBoBgkqhkiG9w0BBwEwHgYJYIZIAWUDBAEuMBEEDEFogLqPWZconQhwHAIBEIA7d9AC7GeJJM34njQvg4Wf1d5sw0NIo1MrBqZa+YdhV8MrkBQPeac0ReRVNDt9qleAt+SHgIRF8P0H+7U=",
+                  "KeyId": "arn:aws:kms:us-east-2:111122223333:key/1234abcd-12ab-34cd-56ef-1234567890ab",
+                  "Plaintext": "VdzKNHGzUAzJeRBVY+uUmofUGGiDzyB3+i9fVkh3piw="
+                }
+                """;
+        var ciphertextBlobBytes = BASE64_DECODER.decode(
+                "AQEDAHjRYf5WytIc0C857tFSnBaPn2F8DgfmThbJlGfR8P3WlwAAAH4wfAYJKoZIhvcNAQcGoG8wbQIBADBoBgkqhkiG9w0BBwEwHgYJYIZIAWUDBAEuMBEEDEFogLqPWZconQhwHAIBEIA7d9AC7GeJJM34njQvg4Wf1d5sw0NIo1MrBqZa+YdhV8MrkBQPeac0ReRVNDt9qleAt+SHgIRF8P0H+7U=");
+        var plainTextBytes = BASE64_DECODER.decode("VdzKNHGzUAzJeRBVY+uUmofUGGiDzyB3+i9fVkh3piw=");
+
+        withMockAwsWithSingleResponse(response, vaultKms -> {
+            var aliasStage = vaultKms.generateDekPair("alias");
+            assertThat(aliasStage)
+                    .succeedsWithin(Duration.ofSeconds(5))
+                    .extracting(DekPair::edek)
+                    .isEqualTo(new AwsKmsEdek("alias", ciphertextBlobBytes));
+
+            assertThat(aliasStage)
+                    .succeedsWithin(Duration.ofSeconds(5))
+                    .extracting(DekPair::dek)
+                    .asInstanceOf(InstanceOfAssertFactories.type(DestroyableRawSecretKey.class))
+                    .matches(key -> SecretKeyUtils.same(key, DestroyableRawSecretKey.takeCopyOf(plainTextBytes, "AES")));
+        });
+    }
+
+    @Test
+    void decryptEdek() {
+        var plainTextBytes = BASE64_DECODER.decode("VGhpcyBpcyBEYXkgMSBmb3IgdGhlIEludGVybmV0Cg==");
+        // response from https://docs.aws.amazon.com/kms/latest/APIReference/API_Decrypt.html
+        var response = """
+                {
+                  "KeyId": "arn:aws:kms:us-west-2:111122223333:key/1234abcd-12ab-34cd-56ef-1234567890ab",
+                  "Plaintext": "VGhpcyBpcyBEYXkgMSBmb3IgdGhlIEludGVybmV0Cg==",
+                  "EncryptionAlgorithm": "SYMMETRIC_DEFAULT"
```

Comment:
Should we be noting in our AWS KMS documentation that `SYMMETRIC_DEFAULT` is the only `EncryptionAlgorithm` we support?

Reply:
Agreed.  A job for https://github.com/kroxylicious/kroxylicious/issues/1216.